source/unified-test-format/unified-test-format.rst

```diff
@@ -1798,6 +1843,134 @@ Use a ``listIndexes`` command to check whether the index exists. Note that it is
+  Each sub-operation must be a valid operation as described in this
+  specification.
+  
+  If, in the course of executing sub-operations, a sub-operation produces
+  an error or a failure, the test runner MUST NOT execute subsequent
+  sub-operations in the same loop iteration. Depending on the presence of
+  ``storeErrorsAsEntity`` and ``storeFailuresAsEntity`` options, as
+  described below, either the iteration MUST complete and the next iteration
+  MUST be started, or the entire ``loop`` operation MUST complete with the
+  produced failure or error.
+
+- ``storeErrorsAsEntity``: if specified, the runner MUST handle errors
+  arising during sub-operation execution and append a document with error
+  information to the array stored in the specified entity. If
+  ``storeFailuresAsEntity`` is specified, the runner MUST NOT include
+  failures in the errors, to the extent the test runner distinguishes
+  errors and failures as further described below.
+  If ``storeFailuresAsEntity`` is not specified,
+  the runner MUST include failures in the errors. The error document
+  MUST contain the following fields:
+  
+  - ``error``: the textual description of the error encountered.
+  - ``time``: the number of (floating-point) seconds since the Unix epoch
+    when the error was encountered.
+
+- ``storeFailuresAsEntity``: if specified, the runner MUST handle failures
+  arising during sub-operation execution and append a document with failure
+  information to the array stored in the specified entity. If
+  not specified, the runner MUST treat failures as errors, and either
+  handle them following the logic described in ``storeErrorsAsEntity``
+  or cause them to terminate execution, if ``storeErrorsAsEntity`` is not
+  specified. The failure document MUST contain the following fields:
+  
+  - ``error``: the textual description of the failure encountered.
+  - ``time``: the number of (floating-point) seconds since the Unix epoch
+    when the failure was encountered.
+
+- ``storeSuccessesAsEntity``: if specfied, the runner MUST keep track of
+  the number of sub-operations that completed successfully, and store
+  that number in the specified entity. For example, if the loop contains
+  two sub-operations, and they complete successfully, each loop execution
+  would increment the number of successes by 2.
+
+- ``storeIterationsAsEntity``: if specfied, the runner MUST keep track of
+  the number of iterations of the loop performed, and store that number
+  in the specified entity.
+
+A *failure* is when an operation performed by the test runner has a
+different result from what the test expected. For example, a test
+containing an expectation that a particular document has a particular value
+in a particular field, and the actual value being different, would
+be a failure. An *error* is any other type of error. For example,
+attempting to invoke an undefined operation would be an error.
```

Comment:
Suggested rephrase:

> A *failure* is when the result or outcome of an operation executed by the test runner differs from its expected outcome. For example, an `expectResult` assertion failing to match a BSON document or an `expectError` assertion failing to match an error message would be considered a failure. An *error* is any other type of error raised by the test runner. For example, an unsupported operation or inability to resolve an entity name would be considered an error.

I originally wrote "failure to resolve an entity name" but didn't want to repeat "failure" there to avoid possible confusion.

Reply:
Added thanks!